test/passes/optimize-instructions_all-features.txt

```diff
@@ -5698,3 +5698,10 @@
   )
  )
 )
+(module
+ (type $i32_=>_i32 (func (param i32) (result i32)))
+ (type $ref_null<i32_->_i32>_=>_i32 (func (param (ref null (func (param i32) (result i32)))) (result i32)))
```

Comment:
Hmm, should this be using `$i32_=>_i32`?

Reply:
I think both are valid, and I'm not sure which is best actually. The benefit of printing the full structure of the type is that we can print a type anywhere in the codebase - that is, without collecting all the types and giving them names etc.

But, with some work maybe we could special-case the type section declarations to be more concise?

Reply:
This seems fine. We can change it if we have to depending on how https://github.com/WebAssembly/function-references/issues/42 is resolved.